provider/azure_test.go

```diff
@@ -176,16 +180,32 @@ func validateAzureEndpoints(t *testing.T, endpoints []*endpoint.Endpoint, expect
 }
 
 func TestAzureRecord(t *testing.T) {
-	zonesClient := mockZonesClient{
-		mockZoneListResult: &dns.ZoneListResult{
-			Value: &[]dns.Zone{
-				createMockZone("example.com", "/dnszones/example.com"),
-			},
+	zlr := dns.ZoneListResult{
+		Value: &[]dns.Zone{
+			createMockZone("example.com", "/dnszones/example.com"),
 		},
 	}
+	results := []dns.ZoneListResult{
+		zlr,
+	}
+
+	mockZoneListResultPage := dns.NewZoneListResultPage(func(ctxParam context.Context, zlrParam dns.ZoneListResult) (dns.ZoneListResult, error) {
+		if len(results) > 0 {
+			result := results[0]
+			results = nil
```

Comment:
does it need to be assigned to nil?

Reply:
It's a hack to escape the iterator, there's probably a nicer way of doing it, before this change was there it had an infinite loop as it was a static list